test/cases/ostree-vsphere.sh

```diff
@@ -0,0 +1,884 @@
+        },
+        "mode": 493
+      }
+    ]
+  },
+  "systemd": {
+    "units": [
+      {
+        "contents": "[Unit]\nDescription=A hello world unit!\n[Service]\nType=oneshot\nRemainAfterExit=yes\nExecStart=/usr/local/bin/startup.sh\n[Install]\nWantedBy=multi-user.target\n",
+        "enabled": true,
+        "name": "hello.service"
+      },
+      {
+        "dropins": [
+          {
+            "contents": "[Service]\nEnvironment=LOG_LEVEL=trace\n",
+            "name": "log_trace.conf"
+          }
+        ],
+        "name": "fdo-client-linuxapp.service"
+      }
+    ]
+  }
+}
+EOF
+
+##################################################################
+##
+## Build edge-vsphere with Ignition firstboot
+##
+##################################################################
+greenprint "📋 Build edge-vsphere image"
+tee "$BLUEPRINT_FILE" > /dev/null << EOF
+name = "vmdk"
+description = "A rhel-edge vmdk image"
+version = "0.0.1"
+modules = []
+groups = []
+
+[customizations.ignition.firstboot]
+url = "${IGNITION_SERVER_URL}/config.ign"
+EOF
+
+greenprint "📄 vmdk blueprint"
+cat "$BLUEPRINT_FILE"
+
+# Prepare the blueprint for the compose.
+greenprint "📋 Preparing installer blueprint"
+sudo composer-cli blueprints push "$BLUEPRINT_FILE"
+sudo composer-cli blueprints depsolve vmdk
+
+# Build simplified installer iso image.
+build_image vmdk "${INSTALLER_TYPE}" "${PROD_REPO_URL}/"
+
+# Download the image
+greenprint "📥 Downloading the vmdk image"
+sudo composer-cli compose image "${COMPOSE_ID}" > /dev/null
+ISO_FILENAME="${COMPOSE_ID}-${INSTALLER_FILENAME}"
+sudo mv "${ISO_FILENAME}" edge-vsphere.vmdk
+
+# Clean compose and blueprints.
+greenprint "🧹 Clean up vmdk blueprint and compose"
+sudo composer-cli compose delete "${COMPOSE_ID}" > /dev/null
+sudo composer-cli blueprints delete vmdk > /dev/null
+
+##################################################################
+##
+## Create vm on datacenter7.0-amd and test it
+##
+##################################################################
+# Datacenter setup
+VSPHERE_DC70="Datacenter7.0-AMD"
+VSPHERE_DC70_HOST="10.73.196.21"
+VSPHERE_DC70_DATASTORE="datastore-21"
+DC70_VSPHERE_VM_NAME="${COMPOSE_ID}-70"
+
+# Remove all existing disk files
+greenprint "🧹 Remove existing vmdk disk files in vsphere datacenter"
+sudo sshpass -p "${DATACENTER_PASSWORD}" ssh "${SSH_OPTIONS[@]}" "${DATACENTER_USERNAME}@${VSPHERE_DC70_HOST}" \
+    rm -fr /vmfs/volumes/${VSPHERE_DC70_DATASTORE}/edge-vsphere
+
+# Upload image to vsphere and create VM
+export GOVC_URL="https://bootp-73-75-144.lab.eng.pek2.redhat.com/sdk"
+export GOVC_USERNAME="${VSPHERE_USERNAME}"
+export GOVC_PASSWORD="${VSPHERE_PASSWORD}"
+export GOVC_INSECURE="true"
+
+greenprint "📋 Uploading vmdk image to vsphere datacenter 7.0"
+export GOVC_DATACENTER="Datacenter7.0-AMD"
+govc import.vmdk -ds="datastore-21" -pool="/Datacenter7.0-AMD/host/Cluster7.0-AMD/Resources" ./edge-vsphere.vmdk
```

Comment:
You can't use a fixed file name edge-vsphere.vmdk in datastore. That will be overwrote by another test running at almost the same time (e.g. two or more PRs are sent almost at the same time)

Reply:
Removed fixed name, using composer id as name.